.github/workflows/check_security_vulnerability.yml

```diff
@@ -1,32 +1,82 @@
-# This workflow uses actions that are not certified by GitHub.
-# They are provided by a third-party and are governed by
-# separate terms of service, privacy policy, and support
-# documentation.
-
-name: Check Security Vulnerability
+name: Code Quality and Security
 
 on:
   pull_request:
   push:
     branches:
       - main
+  schedule:
+    - cron: '0 0 * * 0' # Run every Sunday at 00:00 (midnight)
 
 jobs:
-  lint:
-    name: DevSkim
-    runs-on: ubuntu-latest
+  shared-setup:
+    name: Shared Setup
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
     permissions:
       actions: read
       contents: read
       security-events: write
+    outputs:
+      checkout_ref: ${{ steps.checkout.outputs.ref }}
+      os: ${{ matrix.os }}
     steps:
       - name: Checkout code
+        id: checkout
         uses: actions/checkout@v4
 
+  devskim:
+    name: DevSkim Security Scan
+    needs: shared-setup
+    runs-on: ubuntu-latest
```

Comment:
I saw this was changed in the last commit, why this change?

Reply:
I noticed devSkim scanner failed on macos and windows